theme/static/js/hs_file_browser.js

```diff
@@ -1243,7 +1300,7 @@ $(document).ready(function () {
                 });
 
                 this.on('sending', function (file, xhr, formData) {
-                    formData.append('file_folder', $("#upload-folder-path").text());
+                    formData.append('file_folder', "data/" + $("#upload-folder-path").text());
```

Comment:
What/Where/Why did "data/" pop up in here and above in 1247?  I like to gripe about how we handle /data/contents/ path differently in the system.

Reply:
When sending a request to the API we need to send the full path as it is implemented originally. But since the data folder is not accessible in the file browser we don't need to show it in the UI.

@pkdash I suggest we make the change accordingly to ommit the "data" part from the path. At least for request coming from the UI.

Reply:
You should be able to omit data/contents as it is mostly invisible to the user outside of a bag.

Reply:
@Maurier The back-end view function for file upload should work if the UI doesn't include '/data/contents' as part of the path. I assume you also want the call to refresh file browser for a given path not to include '/data/contents' as part of the path. That needs change in the corresponding view function at the back-end.

Reply:
@pkdash, that's correct. Let's remove the 'data/contents' part from the path for the refresh method.